src/main/java/org/opentripplanner/middleware/utils/NotificationUtils.java

```diff
@@ -157,5 +165,77 @@ public static boolean sendEmailViaSendGrid(String toEmail, String subject, Strin
         }
 
     }
+
+    /**
+     * Ensures that the provided phone number is a domestic mobile number.
+     * Returns a 400-Bad request status if that is not the case.
+     */
+    public static com.twilio.rest.lookups.v1.PhoneNumber ensureDomesticPhoneNumber(spark.Request req, String phoneNumberString) {
```

Comment:
This method needs a lot of work before being ready to ship

- First, I don't think it should take a request as a parameter. Take a look at `ApiGatewayUtils#createApiKey` which throws a `CreateApiKeyException` for a pattern that could be mimicked here.

https://github.com/ibi-group/otp-middleware/blob/692c66afa6e2af09e415154039d870a556ff2260/src/main/java/org/opentripplanner/middleware/utils/ApiGatewayUtils.java#L64

- Also, it appears to contain some strange logic for fake numbers, which could cause issues in a production environment.
- It also uses a Twilio service (lookup API) to perform this lookup. I think a simpler (and less costly) approach would just to use a regex as I described in a previous comment.

Reply:
I see three big advantages for using Twilio's lookup API over a regex, if cost is not too much of a concern:
- No need to configure a regex for each phone number format we encounter (the country code is enough)
- We can accept phone numbers from the UI in any format, and Twilio will convert it into both an E.164 format (+15555550123) to use to send trip monitoring notifications, and a localized format (555) 555-0123 that is sent back to the UI as of writing. That also means we don't have to implement the logic/configurations to strip the extra characters or figure out country prefixes in order to create an E.164 number for notifications.
- We can exclude international numbers from notifications (for US-formatted numbers, that including Canadian and non-US numbers that have the +1 prefix), and potentially landlines.